openedx/core/djangoapps/course_date_signals/tests.py

```diff
@@ -114,5 +116,267 @@ def test_sequence_with_graded_and_ungraded_assignments(self):
+                sequential3 = ItemFactory.create(category='sequential', parent=chapter)
+            self.courses.append(course3)
+
+            # course 3: a section with a subsection that has due_num_weeks and a section without due_num_weeks that has graded content
+            course4 = CourseFactory.create(self_paced=True)
+            with self.store.bulk_operations(course4.id):
+                chapter1 = ItemFactory.create(category='chapter', parent=course4)
+                sequential1 = ItemFactory.create(category='sequential', parent=chapter1, due_num_weeks=2)
+                vertical1 = ItemFactory.create(category='vertical', parent=sequential1)
+                ItemFactory.create(category='problem', parent=vertical1)
+
+                chapter2 = ItemFactory.create(category='chapter', parent=course4)
+                sequential2 = ItemFactory.create(category='sequential', parent=chapter2, graded=True)
+                vertical2 = ItemFactory.create(category='vertical', parent=sequential2)
+                ItemFactory.create(category='problem', parent=vertical2)
+            self.courses.append(course4)
+
+            # course 4: a section with a subsection that has due_num_weeks and multiple sections without due_num_weeks that has graded content
+            course5 = CourseFactory.create(self_paced=True)
+            with self.store.bulk_operations(course5.id):
+                chapter1 = ItemFactory.create(category='chapter', parent=course5)
+                sequential1 = ItemFactory.create(category='sequential', parent=chapter1, due_num_weeks=2)
+                vertical1 = ItemFactory.create(category='vertical', parent=sequential1)
+                ItemFactory.create(category='problem', parent=vertical1)
+
+                chapter2 = ItemFactory.create(category='chapter', parent=course5)
+                sequential2 = ItemFactory.create(category='sequential', parent=chapter2, graded=True)
+                vertical2 = ItemFactory.create(category='vertical', parent=sequential2)
+                ItemFactory.create(category='problem', parent=vertical2)
+
+                chapter3 = ItemFactory.create(category='chapter', parent=course5)
+                sequential3 = ItemFactory.create(category='sequential', parent=chapter3, graded=True)
+                vertical3 = ItemFactory.create(category='vertical', parent=sequential3)
+                ItemFactory.create(category='problem', parent=vertical3)
+
+                chapter4 = ItemFactory.create(category='chapter', parent=course5)
+                sequential4 = ItemFactory.create(category='sequential', parent=chapter4, graded=True)
+                vertical4 = ItemFactory.create(category='vertical', parent=sequential4)
+                ItemFactory.create(category='problem', parent=vertical4)
+            self.courses.append(course5)
+
+    @override_waffle_flag(CUSTOM_PLS, active=True)
+    def test_extract_dates_from_course_inheritance(self):
```

Comment:
I appreciate your thoroughness with this test, but I think for simplicity, you could just fast forward straight to the problem case and show everything is inherited all at once rather than iteratively